packages/client/src/pages/Trecipe/TrecipePage.tsx

```diff
@@ -255,12 +273,46 @@ class TrecipePage extends React.Component<TrecipeProps, TrecipeState> {
         };
     }
 
+    private renderDestinationList(completed: Set<string>) {
+        return (
+            <DragDropContext onDragEnd={this.onDestDragEnd.bind(this)}>
+                <Droppable droppableId="droppable">
+                    {(provided) => (
+                        <ul
+                            className="destination-cards"
+                            {...provided.droppableProps}
+                            ref={provided.innerRef}>
+                            {this.getDestinationsList().map((dest, index) => (
+                                <Link
+                                    className="router-link"
+                                    to={`/destinations/${dest.placeId}`}
+                                    target="_blank"
+                                    key={dest.uuid}>
+                                    <DestinationCard
+                                        key={dest.uuid}
+                                        destination={dest}
+                                        isCompleted={completed.has(dest.uuid)}
+                                        index={index}
+                                        onClickDelete={this.onDestDeleteClick.bind(this)}
+                                        onClickComplete={this.onDestCompleteClick.bind(this)}
+                                        isInEdit={this.state.isInEdit}
+                                    />
+                                </Link>
+                            ))}
+                            {provided.placeholder}
+                        </ul>
+                    )}
+                </Droppable>
+            </DragDropContext>
+        );
+    }
+
     render() {
         const trecipe: Trecipe | undefined = this.props.trecipe;
         const destinations: Destination[] | undefined = this.props.destinations;
         const editTrecipeBtnString = 'Edit Trecipe';
         if (!trecipe || !destinations) {
-            return null;
+            return <FullScreenLoader />;
         } else {
```

Comment:
> Those spinners and error handling looks amazing! I kinda wanna refactor the login page to use those spinner too lol
> Didn't look at every single line of code, but here are some minor changes:
> 
> I think there is a bit too much white space in between destination and the static map when there is only one destination in a Trecipe:
> ![Screenshot_2020-08-04 Trecipe](https://user-images.githubusercontent.com/36020702/89362731-775f1000-d683-11ea-94a6-c9f725ce5f8b.png)

![image](https://user-images.githubusercontent.com/33335963/89384178-b44ef500-d6cb-11ea-93e6-d10b121a97dc.png)

A little better? hahaha. Reduced 80vh to 50vh.

Reply:
> If trecipe or destination failed to load, I think the full screen loader will be there forever?

Yeaaa right now, since the `errorReducer` is not used by any components, we're not checking for 404 from server... Just displays a toaster saying it failed.

Reply:
> A little better? hahaha. Reduced 80vh to 50vh.

Yea that is better thank you! 

> Yeaaa right now, since the errorReducer is not used by any components, we're not checking for 404 from server... Just displays a toaster saying it failed.

Oic, since there is toaster, it should be okay to ignore this for now 😅